pydantic/utils.py

```diff
@@ -136,6 +139,49 @@ def almost_equal_floats(value_1: float, value_2: float, *, delta: float = 1e-8)
     return abs(value_1 - value_2) <= delta
 
 
+def generate_model_signature(
+    init: Callable[..., None], fields: Dict[str, 'ModelField'], config: Type['BaseConfig']
+) -> Signature:
+    """
+    Generate signature for model based on its fields
+    """
+
+    present_params = signature(init).parameters.values()
+    merged_params: Dict[str, Parameter] = {}
+    var_kw = None
+    use_var_kw = False
+
+    for param in islice(present_params, 1, len(present_params)):  # skip self arg
```

Comment:
You can `islice(present_params, 1, None)` to take to end-of-iterable, which avoids readers having to think about indexing semantics.

Reply:
Good point, thank you! Would you like to suggest the change?

Reply:
Nah, go ahead - I appreciate the offer but don't need credit 🙂